go.mod

```diff
@@ -4,7 +4,7 @@ go 1.17
 
 require (
 	github.com/dustinkirkland/golang-petname v0.0.0-20191129215211-8e5a1ed0cff0
-	github.com/hashicorp/consul/api v1.11.0
+	github.com/hashicorp/consul/api v1.10.1-0.20221011220032-d17af23641d4
```

Comment:
Was this intentional? Should we be downgrading the `api` dependency?

Reply:
yes, api v1.11.0 is an old release api version. Since we need some new field in the query option, like `Peer`, I run `go get github.com/hashicorp/consul/api@d17af23641d41fb7cfa56884711d071cf54b2a0d`, which is a recent git sha, which results in this line.

Actually when I check consul-k8s [repo](https://github.com/hashicorp/consul-k8s/blob/2d0207e6ec95a03d8d45c889c3f080c1a5b53e2e/control-plane/go.mod#L13), it has similar sha; so I guess it should be fine.